test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/exception/CborExceptionTest.java

```diff
@@ -0,0 +1,275 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.awscore.exception.AwsErrorDetails;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.protocolcbor.ProtocolCborClient;
+import software.amazon.awssdk.services.protocolcbor.model.AllTypesRequest;
+import software.amazon.awssdk.services.protocolcbor.model.EmptyModeledException;
+import software.amazon.awssdk.services.protocolcbor.model.ImplicitPayloadException;
+import software.amazon.awssdk.services.protocolcbor.model.ProtocolCborException;
+import software.amazon.awssdk.thirdparty.jackson.dataformat.cbor.CBORFactory;
+import software.amazon.awssdk.thirdparty.jackson.dataformat.cbor.CBORGenerator;
+import software.amazon.awssdk.utils.MapUtils;
+
+/**
+ * Exception related tests for CBOR.
+ */
+public class CborExceptionTest {
+    private static final String PATH = "/";
+
+    @Rule
+    public WireMockRule wireMock = new WireMockRule(0);
+
+    private ProtocolCborClient client;
+
+    private CBORFactory cborFactory;
+
+    @Before
+    public void setupClient() {
+        client = ProtocolCborClient.builder()
+                                   .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("akid", "skid")))
+                                   .region(Region.US_EAST_1)
+                                   .endpointOverride(URI.create("http://localhost:" + wireMock.port()))
+                                   .build();
+
+        cborFactory = new CBORFactory();
+    }
+
+    @Test
+    public void unmodeledException_UnmarshalledIntoBaseServiceException() throws IOException {
+        Map<?, ?> obj = MapUtils.of(
+            "__type", "SomeUnknownType"
+        );
+
+        stub404Response(PATH, toCbor(obj));
+        assertThatThrownBy(() -> client.allTypes(AllTypesRequest.builder().build()))
+            .isExactlyInstanceOf(ProtocolCborException.class);
+    }
+
+    @Test
+    public void modeledExceptionWithImplicitPayloadMembers_UnmarshalledIntoModeledException() throws IOException {
+        Map obj = new HashMap();
+        obj.put("__type", "ImplicitPayloadException");
+        obj.put("StringMember", "foo");
+        obj.put("IntegerMember", Integer.valueOf(42));
+        obj.put("LongMember", Long.valueOf(9001));
+        obj.put("DoubleMember", Double.valueOf(1234.56));
+        obj.put("FloatMember", Float.valueOf(789.10f));
+        obj.put("TimestampMember", Long.valueOf(1398796238123L));
```

Comment:
note: this bit [differs from `AwsJsonExceptionTest`](https://github.com/aws/aws-sdk-java-v2/blob/eeee2389d4d1f0d776ac75e492469dea9c817a0e/test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/exception/AwsJsonExceptionTest.java#L80). Seems like in CBOR we unmarshall from UNIX_TIMESTAMP_MILLIS, rather than UNIX_TIMESTAMP

Reply:
https://github.com/aws/aws-sdk-java-v2/blob/eeee2389d4d1f0d776ac75e492469dea9c817a0e/core/protocols/aws-cbor-protocol/src/main/java/software/amazon/awssdk/protocols/cbor/AwsCborProtocolFactory.java#L99